### ISSUE TITLE
Improve consistency of hint text messages for form components

### ISSUE THREAD

**Issue by k80bowman** [department-of-veterans-affairs/vets-design-system-documentation#1249]
## Description
As part of his review of department-of-veterans-affairs/component-library#535, @bkjohnson noted that we have some [inconsistencies](https://github.com/department-of-veterans-affairs/component-library/pull/535#pullrequestreview-1124478430) with how we we manage hint text/descriptions in our form components.

The color of the hint text in our components also does not meet color contrast guidelines and we need to fix that. If we could have one hint text class that gets used by all the form components (and imported into their CSS), it would probably be better than trying to manage the styles individually.

## Tasks
- [x] Work with @GnatalieH and @humancompanion-usds to determine the best path forward for form component hint text
- [x] Implement the solution
- [ ] Fix the hint text color

## Acceptance Criteria
- [x] Form component hint text implementation is consistent
- [x] Hint text color meets accessibility color contrast guidelines

**Comment by humancompanion-usds** [comment]
I spoke with DK about the hint-text color and it does pass AA, just not AAA. He assured me that was acceptable and we agreed that some hierarchy is needed given how busy a text field can potentially become with label, additional info, hint text, error message, etc.

**Comment by jamigibbs** [comment]
## Overview

| Component  | "Hint" Type |
| ------------- | ------------- |
| va-checkbox  | prop (named "description") and slot (named "description")  |
| va-date  | default slot  |
| va-memorable-date  | default slot  |
| va-file-input  | default slot  |
| va-number-input  | none  |
| va-radio (v1)  | prop (named "hint")  |
| va-radio-option (v1)  | none (to be added #1373)  |
| va-select  | none  |
| va-text-input  | default slot  |
| va-textarea  | none  |

## Details

### [Checkbox](https://design.va.gov/storybook/?path=/docs/components-va-checkbox--with-description-string)

The `va-checkbox` component allows any number of "description" text to be passed into the component. The component accepts the description as either:

a) a `description` prop
b) and/or a `description` slot

The text color is currently #323a45 (`--color-gray-dark`). The description is displayed in a `p` element.

![Screen Shot 2022-12-07 at 10 13 22 AM](https://user-images.githubusercontent.com/872479/206231868-779a70b7-a212-4a88-93ee-20aee33a60e2.png)

### [Date Input](https://design.va.gov/storybook/?path=/docs/components-va-date--with-hint-text)

The `va-date` component has text that is passed into the component from the default slot. We are describing this as "hint text" in the storybook example but the code does not make any distinction here. It's simply the `<slot />`.

A default color is set on the component host and is `--color-gray` (#5b616b) which is what the slot text is displaying.
All font sizes are 1.6rem (12px).

![Screen Shot 2022-12-07 at 10 41 47 AM](https://user-images.githubusercontent.com/872479/206238429-47367b3a-3592-4c69-95ad-a5168f113636.png)

### [Memorable Date](https://design.va.gov/storybook/?path=/docs/components-va-memorable-date--with-hint-text-error)

The `va-memorable-date` component has a dynamic hint text that is passed into the component from the default slot. We are describing this as "hint text" in the storybook example but the code does not make any distinction here. It's simply the `<slot />`.

There is also a predefined "dateHint" text that describes how to use the component inputs. This text is not customizable by props or slots.

The color is set on the component host and is `--color-gray` (#5b616b).
All font sizes are 1.6rem (12px).

![Screen Shot 2022-12-07 at 1 34 33 PM](https://user-images.githubusercontent.com/872479/206278415-431caa23-4217-42c5-88b4-5688d15393c3.png)

### [File Input](https://design.va.gov/storybook/?path=/docs/components-va-file-input--error-message)

The `va-file-input` component does not explicitly display hint text in Storybook but within the code it does have a default  slot positioned between the label and error message.

We may want to add a "Hint Text" example in Storybook that displays the slot content similar to the [va-date example](https://design.va.gov/storybook/?path=/docs/components-va-date--with-hint-text).

<img width="463" alt="Screen Shot 2022-12-07 at 2 19 30 PM" src="https://user-images.githubusercontent.com/872479/206287064-a55fd61a-1c81-4afd-b9e4-c1865de35833.png">

### [Number Input](https://design.va.gov/storybook/?path=/docs/components-va-number-input--error)

The `va-number-input` component does not have any prop or slot to allow for hint text similar to `va-file-input` above.

<img width="559" alt="Screen Shot 2022-12-07 at 2 24 25 PM" src="https://user-images.githubusercontent.com/872479/206287971-728adbce-5f2d-405b-95c2-24fffb5a2bf4.png">

### [Radio Button](https://design.va.gov/storybook/?path=/docs/components-va-radio--hint)

The `va-radio` (v1) component accepts hint text from a prop. The default slot is used for displaying the related `va-radio-option`'s.

<img width="433" alt="Screen Shot 2022-12-07 at 2 34 48 PM" src="https://user-images.githubusercontent.com/872479/206289812-d5028397-b3e6-4f76-9657-3af08ef7a9c0.png">

### [Select](https://design.va.gov/storybook/?path=/docs/components-va-select--error-message)

The `va-select` component does not have a hint text option available (no dedicated prop or slot). The default slot is used for displaying the options.

<img width="599" alt="Screen Shot 2022-12-07 at 2 40 33 PM" src="https://user-images.githubusercontent.com/872479/206291010-2d8b121d-e659-4ba8-94da-78842ee3256a.png">

### [Text Input](https://design.va.gov/storybook/?path=/docs/components-va-text-input--with-hint-text)

The `va-text-input` component displays hint text from a default slot. In our storybook example, we are using `va-additional-info` in the slot but this could technically be any HTML since it's simply passing into a default slot.

<img width="750" alt="Screen Shot 2022-12-07 at 2 51 50 PM" src="https://user-images.githubusercontent.com/872479/206293038-2000e09e-206c-4e1d-8694-251ef2b83060.png">

### [Textarea](https://design.va.gov/storybook/?path=/docs/components-va-textarea--error)

The `va-textarea` component does not have a hint text option available (no dedicated prop or slot). The default slot is used for displaying the options.

<img width="632" alt="Screen Shot 2022-12-07 at 2 58 01 PM" src="https://user-images.githubusercontent.com/872479/206294317-0206e0ba-80db-4d7f-ba5b-e05727f5c0ac.png">

## Next steps/actions

- [ ] Make all label and hint/description text consistent across all of the form components above. This means that labels should be `--color-base` (#212121) and hint/descriptions should be styled to follow what is outlined on the [form documentation page](https://design.va.gov/components/form/#hint-text).
- [ ] Improve hierarchy between label, hint/description text, and error message as needed.
- [ ] Decide how all of these components can consistently implement a hint/description text.
- [ ] Add hint text examples to Storybook where there exists the option in a component but it's not shown (ie. va-file-input)

**Comment by jamigibbs** [comment]
Natalie and I were able to sync up and we had a few take-aways after reviewing the info above:

1. Add a "hint" prop to all form components (with the exception of va-text-input -- see below) - We should make the hint text mechanism consistent across all of these components. This means adding a prop `hint` to components that don't have one yet so that a single string could be passed in for managing its style consistently. This would be the same thing that we did for `va-radio` recently: https://github.com/department-of-veterans-affairs/component-library/pull/535.

2. Rename the `va-text-input` storybook example from "With Hint Text" to "With Additional Info" - This component would be the one exception to having a `hint` prop and we would not add one here (we thought it was unnecessary in this case and it would make the component content too busy). @humancompanion-usds Can you confirm that for this component we should leave it as-is? We briefly discussed doing some kind of conditional display where if a hint prop was populated, it would not display the additional info slot but we weren't convinced that approach was needed or necessary.

![Screen Shot 2022-12-12 at 3 10 24 PM](https://user-images.githubusercontent.com/872479/207155268-546c8865-cabd-4df4-b926-d736ac30c48e.png)

4. The hint text will use the same color treatment as the USWDS class `.usa-hint`: #71767A

3. Make sure all of the form component label text, hint text, and error text have consistent spacing. This means 12px based on USWDS.

4. Where label and input text color is not displaying as `--color-base` #212121, make that update.

cc @GnatalieH

**Comment by humancompanion-usds** [comment]
We resolved with DK that our hint-text does meet contrast requirements. The hierarchy of color should also be helped by the fact that we fixed our base color to be correct (#212121) where it had been a shade of dark gray.
